edx_proctoring/views.py

```diff
@@ -286,6 +286,72 @@ def get(self, request, exam_id=None, course_id=None, content_id=None):  # pylint
         return Response(data)
 
 
+class StudentOnboardingStatusView(ProctoredAPIView):
+    """
+    Endpoint for the StudentOnboardingStatusView
+
+    Supports:
+        HTTP GET: returns the learner's onboarding status relative to the given course_id
+
+    HTTP GET
+        /edx_proctoring/v1/user_onboarding/status?course_id=edX/DemoX/Demo_Course
+
+    **Response Values**
+        * 'onboarding_status': string specifying the learner's onboarding status;
+            will return NULL if there are no onboarding attempts
+        * 'onboarding_link': link to the onboarding exam
+    """
+    def get(self, request):
+        """
+        HTTP GET handler. Returns the learner's onboarding status.
+        """
+        course_id = request.GET.get('course_id')
```

Comment:
If course_id is required on the view, you should incorporate it into the URL. But is course_id required? Can the API return a dictionary of all the onboarding status of all the courses the learner is with? Also, I thought onboarding status within Verificient should be course independent, meaning, if a user got onboarding approved for course A, they are also approved for course B, right?
If course_id is not required, I would agree you should keep that as the query string parameter, but you would need better code handling for no course_id case, and I want to see that covered by your unit tests.

Reply:
Yes onboarding status with Verificient doesn't have to be course dependent but I think we should still treat it that way on our end right now. The user experience for onboarding is still tied to a single course and I think allowing that state to carry over between courses may lead to some confusion. (ex. a rejected onboarding in a different course) However, I like this endpoint design because if we go down that path in a next iteration we don't have to create a new endpoint, we can just drop the query parameter.

Simplest solution IMO is just have `course_id` be a mandatory query parameter right now. (and a test covering this)

Reply:
I left the course_id as a required query parameter, but also added logic so that it can be easily removed as a requirement in the future (IE if a course_id isn't given, the onboarding link will just be returned as None)